### ISSUE TITLE
Add better documentation for system tests

### ISSUE THREAD

**Issue by aried3r** [rspec/rspec-rails#1902]
Hey! I was really happy to see [RSpec 3.7](http://rspec.info/blog/2017/10/rspec-3-7-has-been-released/) was released with support for [ActionDispatch::SystemTest](http://guides.rubyonrails.org/testing.html#system-testing). Great work!

However I'm not sure what the recommended way of using them is. The [README](https://github.com/rspec/rspec-rails/blob/c543765e422b6a0e6c31a3fed383752e8a12efbc/README.md) has no mention of them but in [one of the commits](https://github.com/rspec/rspec-rails/commit/2fb410b2670fdaf22f6bd2d0b8d67f9d1b78334d#diff-4605835d2b89c95ec57c38eca0c807f3R3) there's a small note about usage.

One of the first questions I had were:
– Can I do something similar to having [`application_system_test_case.rb`](http://guides.rubyonrails.org/testing.html#changing-the-default-settings)? The note in the commit only says it's not used and that you have to call it manually in a test case. What if I want a global config for `type: :system` specs?
– Are the other helpers like the screenshot helper included ([it appears so](https://github.com/rspec/rspec-rails/commit/2fb410b2670fdaf22f6bd2d0b8d67f9d1b78334d#diff-779a18a53423de804d5cbdc1edc6552dR28)? As in, is RSpec really just a thin wrapper around the system tests and besides not being able to use `application_system_test_case.rb` everything works as expected and I just need to follow the Rails docs?
– Speaking of thin wrappers, when Rails 5.2 is released, will using [`:chrome_headless`](https://github.com/rails/rails/pull/30876) just work as a driver or will there have to be a change and new release of rspec-rails as well?

Since a lot of work went into system tests, I just want to make sure I don't miss out on it when using RSpec as a wrapper around them. :)

**Comment by JonRowe** [comment]
> – Can I do something similar to having application_system_test_case.rb? The note in the commit only says it's not used and that you have to call it manually in a test case. What if I want a global config for type: :system specs?

Yes, RSpec has it's own configuration which has existed since long before system tests, you should just be able to do the equivalent in `before(:suite)` hook.

> Are the other helpers like the screenshot helper included (it appears so? As in, is RSpec really just a thin wrapper around the system tests and besides not being able to use application_system_test_case.rb everything works as expected and I just need to follow the Rails docs?

They should do, `application_system_test_case.rb ` just doesn't make sense in RSpecs world.

> Speaking of thin wrappers, when Rails 5.2 is released, will using :chrome_headless just work as a driver or will there have to be a change and new release of rspec-rails as well?

I'd assume so, given we have nothing to do with driver selection.

**Comment by Xosmond** [comment]
So, system tests doesn't matter on RSpecs, Do we still need to use DatabaseCleaner?

**Comment by pachonk** [comment]
@Xosmond Tests are ran in transactions now so you shouldn't need DB cleaner

**Comment by mandysimon88** [comment]
We are getting a ton of intermittent failures with system tests, latest Rspec-Rails and latest Rails.

Documentation would be extremely helpful. Lots of SO answers saying "you shouldn't need DB cleaner" aren't helpful... We shouldn't need it **if we've configured things correctly**. What is correctness?

Hence, need for documentation.

**Comment by JonRowe** [comment]
The docs for system testing are here: `http://guides.rubyonrails.org/testing.html#system-testing` but if someone would like to include some better docs within rspec's own I'll happily review a PR. Remember rspec-rails does not provide or configure this functionality, just exposes it.

**Comment by Xosmond** [comment]
@JonRowe I think I can do that the next week, because this project its `rspec-rails` not `rspec`.

PD: My old problem was the multi tenant gem that doesn't fit good on system tests with rspec.

**Comment by ooffeerr** [comment]
Hi! Is this issue still up for grabs?

**Comment by JonRowe** [comment]
As far as I know!

**Comment by samphippen** [comment]
https://github.com/rspec/rspec-rails/pull/2052 If you're going to add further documentation, this PR is a good place to start!

**Comment by andrewshawcare** [comment]
With `ApplicationSystemTestCase` I can turn off screenshots by doing this: https://stackoverflow.com/a/44337280

If RSpec does not use `ApplicationSystemTestCase`, how do I turn off screenshots?

**Comment by JonRowe** [comment]
You can define the same method within a spec to achieve that, e.g.
```Ruby
module DisableScreenshots
  def take_screenshot
    false
  end
end
RSpec.configuration.include DisableScreenshots, type: :system
```
